util/codecov-ci.sh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+#
+# Copyright Amazon.com Inc. or its affiliates.  All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR ISC
+#
+
+set -xe
+
+SRC=$(pwd)
+
+# Sanity check
+DIRNAME=$(basename -- "${SRC}")
+if [[ "${DIRNAME}" != 'aws-lc' ]]; then
+  echo "Script must be executed from aws-lc directory"
+  exit 1
+fi
+
+BUILD="$1"
+if [ -n "$BUILD" ]; then
+  if [ ! -e "$BUILD" ]; then
+    mkdir -p "$BUILD"
+  fi
+  BUILD=$(readlink -f "$BUILD")
+else
+  echo "Must specify a build directory."
+  exit 1
+fi
+
+cmake -DGCOV=1 -DDISABLE_PERL=1 -DBUILD_TESTING=1 -DBUILD_LIBSSL=1 -DCMAKE_BUILD_TYPE=Debug -S "${SRC}" -B "${BUILD}"
+cmake --build "${BUILD}" --target all_tests
+cmake --build "${BUILD}" --target run_tests
+
+#TODO: Use callgrind
```

Comment:
what's blocking us from using this now?

Reply:
The PR provides test coverage data in our CI for C/C++ code that runs on Linux/x86-64, but it lacks data on any other combination of language, OS, and CPU. There are tools available (like Callgrind) that provide coverage for assembly code.  We should also extend this script to collect data for other CPU architectures.

But for now, I feel like this is a good "milestone" as this change quickly provides important test coverage information for code that is frequently updated (i.e., CPU/OS-agnostic C/C++ code).